### ISSUE TITLE
BoneCPDataSourceProvider unable to set the int provider values partitionCount , maxConnectionsPerPartition

### ISSUE THREAD

**Issue by newgeekorder** [vert-x3/vertx-jdbc-client#116]
### Version

* vert.x core: 3.5.0
* vert.x jdbc-client: 3.5.0
* jdbc driver: org.h2
* RDBMS: h2

### General info

* What is your jdbc driver 
org.h2.JdbcDriver
* What is your RDBMS server
h2
* What is your connection string (no user/passwords please!)
`  h2Client = JDBCClient.createShared(vertx, new JsonObject()
                .put("provider_class", "io.vertx.ext.jdbc.spi.impl.BoneCPDataSourceProvider")
                .put("jdbcUrl", "jdbc:h2:tcp://localhost/~/wcs;MODE=DB2")
                .put("username", "root")
                .put("password", "xxxxx")
//                .put("driver_class", "org.h2.Driver")
                .put("partitionCount", 4)
                .put("maxConnectionsPerPartition", 4));
`
### Context

I encountered an exception which looks suspicious while ...
connecting to the database with BoneCP 
* first through an exception that maxConnectionPool is null 
* debugging the value the code takes the config and calls "set"config with value . The value's passed are json and appear to the setter as Integers. BoneCP uses in in the method signature so no method is found.

If useful I can create a pull request "if Integer" and no method found try "int" 

### Do you have a reproducer?

* Link to github project/gist

### Steps to reproduce

1. ... create a BoneCp connection string that contains int paramters 
2. ... attempt to connect to a database
3. ... A warning will be seen for the config not mapping 
4. ...

### Extra

* Anything that can be relevant

**Comment by tsegismont** [comment]
@newgeekorder yes please provide a pull request with a test which fails without the fix and passes with it
